R/ard_compare.R

```diff
@@ -0,0 +1,193 @@
+#' Compare ARDs
+#'
+#' @include import-standalone-checks.R
+#'
+#' @description
+#' `ard_compare()` compares the `stat`, `fmt_fun`, `warning`, and `error`
+#' columns of two ARDs row-by-row using a shared set of key columns. Rows where
+#' the column values differ are returned.
+#'
+#' @param x (`card`)\cr
+#'   first ARD to compare.
+#' @param y (`card`)\cr
+#'   second ARD to compare.
+#' @param key_columns (`character` or [`tidy-select`][dplyr::dplyr_tidy_select])\cr
+#'   optional specification of column names identifying unique records. Supply a
+#'   character vector or tidyselect expression. When `NULL`, grouping columns
+#'   along with `"variable"`, `"variable_level"`, and `"stat_name"` are used.
+#'   If the ARDs do not have identical primary key sets, the comparison falls
+#'   back to the shared primary key columns provided they uniquely identify rows
+#'   in both inputs.
+#'
+#' @return a named list of data frames containing key columns with the
+#'   corresponding values from both ARDs for rows where the column values do not
+#'   match. The list contains entries for the `stat`, `fmt_fun`, `warning`, and
+#'   `error` columns (the `fmt_fun` component is labeled `fmt_fn` for
+#'   compatibility with previous naming conventions).
+#'
+#' @details
+#' When both ARDs retain metadata about the environment that created them (for
+#' example via an attribute storing an environment object), the function ensures
+#' that the environments match before comparing values.
+#' @importFrom rlang env_label
+#' @export
+#'
+#' @examples
+#' ard_base <- ard_summary(ADSL, variables = AGE)
+#' ard_modified <- ard_summary(dplyr::mutate(ADSL, AGE = AGE + 1), variables = AGE)
+#'
+#' ard_compare(ard_base, ard_modified)$stat
+#'
+ard_compare <- function(x, y, key_columns = NULL) {
```

Comment:
All of our functions that begin with `ard_*()` create new ARDs. Let's name the function `compare_ard()`.